File: src/frontend/src/model/cocktail.ts
import Tag from "./tag";
import Ingredient from "./ingredient";
import Recipe from "./recipe";

export default class Cocktail {
  id: number;
  name: string;
  description: string;
  tags: Tag[];
  ingredients: Ingredient[];
  recipe: Recipe;
  note: string;

  constructor(
    id: number,
    name: string,
    description: string,
    tags: Tag[],
    ingredients: Ingredient[],
    recipe: Recipe,
    note: string
  ) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.tags = tags;
    this.ingredients = ingredients;
    this.recipe = recipe;
    this.note = note;
  }
}

export interface CocktailDTO {
  id: string;
  name: string;
  category: string;
  subcategory: string;
}

File: bartenderly-server/src/main/java/io/github/mityavasilyev/bartenderlyserver/cocktail/Cocktail.java
package io.github.mityavasilyev.bartenderlyserver.cocktail;

import io.github.mityavasilyev.bartenderlyserver.extra.Ingredient;
import io.github.mityavasilyev.bartenderlyserver.extra.Recipe;
import io.github.mityavasilyev.bartenderlyserver.tag.Tag;
import lombok.*;

import javax.persistence.*;
import java.util.Set;

/**
 * Cocktail class that contains Name, Description, Tags,
 * Ingredients, Recipe and Note
 */
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@ToString(includeFieldNames = false)
@Entity(name = "cocktail")
@Table(name = "cocktails")
public class Cocktail {

    @Id
    @GeneratedValue(
            strategy = GenerationType.IDENTITY
    )
    @Column(updatable = false)
    private Long id;

    @Column(
            name = "name",
            nullable = false,
            columnDefinition = "TEXT"
    )
    private String name;

    @Column(
            name = "description",
            columnDefinition = "TEXT"
    )
    private String description;

    @ManyToMany
    @JoinTable(
            name = "tags_cocktails",
            joinColumns = @JoinColumn(
                    name = "cocktail_id",
                    referencedColumnName = "id"
            ),
            inverseJoinColumns = @JoinColumn(
                    name = "tag_id",
                    referencedColumnName = "id"
            )
    )
    private Set<Tag> tags;

    @OneToMany(
            cascade = CascadeType.ALL
    )
    @JoinColumn(
            name = "cocktail_id",
            referencedColumnName = "id"
    )
    private Set<Ingredient> ingredients;

    @OneToOne(
            cascade = CascadeType.ALL
    )
    private Recipe recipe;

    @Column(
            name = "note",
            columnDefinition = "TEXT"
    )
    private String note;

}
